docs: add layer styleguide

amstewart opened the pull request:
Add a first draft of a meta-nilrt layer styleguide, to coordinate syntax
and styling.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

This styleguide includes some existing style sections from the [nilrt.git CONTRIBUTING guide](https://github.com/ni/nilrt/blob/a8e494d6181a7f858e0cd4385c68e40c0a7e06d3/docs/CONTRIBUTING.md?plain=1#L41). I'll submit a PR to remove it from that document.

amstewart committed:
docs: add layer styleguide

Add a first draft of a meta-nilrt layer styleguide, to coordinate syntax
and styling.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

bstreiff commented on docs/styleguide.md:
```diff
@@ -0,0 +1,198 @@
+
+* `Upstream-Status: Inappropriate [$rationale]` ([ex](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-support/curl/curl/0005-Add-nicurl-wrapper-functions.patch)) For when the `.patch` change is specific to NI LinuxRT and would not be desired (or has been rejected) by upstream.
+* `Upstream-Status: Not Submitted [$rationale]` ([ex](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-gnome/florence/files/0004-Add-option-for-automatic-bring-to-top.patch)) For when the `.patch` file is being included in NILRT before being submitted to its upstream project. This should only occur if the `.patch` is needed for an immediate NILRT release and there is no time to get ...
+* `Upstream-Status: Submitted [$upstream_mailing_list_link]` ([ex](https://github.com/ni/meta-nilrt/blob/1e6453ebca8735de96eaaf3bc931d22998c8dfb3/recipes-support/curl/curl/0014-Fixup-lib1529-test.patch)) For when the `.patch` has been submitted to the upstream project's mailing list, but needs to be pulled into NILRT prior to final upstream approval.
```
nit: some upstream projects don't use mailing lists, patch submissions happen via a web PR system. Would we have (for example) a GitHub or GitLab PR URL as the "upstream mailing list link" in that situation?

amstewart replied:
Yes. I'll add some clarification to the doc.

amstewart committed:
styleguide: accomidate upstreams without mailing lists

Not all upstream projects have "mailing lists", and sections of the
styleguide which reference linking to the upstream list might be
confusing to developers.

Make it clear that the upstream link can refer to a merge PR as well.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

amstewart commented:
# Patch V2
* rebased to the hardknott mainline
* [Implemented feedback](https://github.com/ni/meta-nilrt/pull/447/commits/fc8dc83397d8a24bd7e83e76ca9b7d5d449010c7) from @bstreiff .

bstreiff reviewed (approved):
"accomidate upstreames" could use a spellcheck pass but I assume you'll probably squash that into the other commit.

shruthi-ravi commented on docs/styleguide.md:
```diff
@@ -0,0 +1,198 @@
+
+**Indentation**
+
+Use **tabs** for indentation and **spaces** for alignment in all places - *except* within python task functions, where spaces should be used for both.
```
Why the deviation here?

amstewart replied:
Because I believe strongly that soft tabs are degenerate, and also upstream isn't actually consistent about using soft tabs.

shruthi-ravi replied:
It sounds like a personal opinion. Even if upstream doesn't enforce the soft tabs, if it is in their style guide, I think we should follow it. It'll be less of a hassle for developers who follow the style guide and wish to contribute to multiple layers.

amstewart replied:
Like I said, upstream doesn't actually enforce this style and you can find many examples of both in upstream. Is this a blocker for you, because I feel strongly about it, but I don't want to go through the effort of selling it to upstream to change on the wiki?

shruthi-ravi replied:
Sorry, I missed this response. I do feel strongly about enforcing this policy since I think it is not sustainable. I think we should only care about consistency within a file or write a linter to do the work for us. But since you've already made all of those changes based on what you felt, I'm not going to block the way.

shruthi-ravi commented on docs/styleguide.md:
```diff
@@ -130,9 +130,9 @@ Contributors are free to use a single newline to create logical breaks between d
 At a minimum, you should add an additional message trailer declaring the [upstream status](https://www.openembedded.org/wiki/Commit_Patch_Message_Guidelines#Patch_Header_Recommendations:_Upstream-Status) of the .patch file at the time of your OE commit. These status lines are crucial to helping the project maintainers properly upgrade and rebase recipes. Common status lines are:
```
Typos in the commit message.

amstewart replied:
Fixed.

amstewart commented:
# Patch V3
* Audited the commit messages and styleguide body for spelling and grammar.
